### ISSUE TITLE
[bug] Download speed problem

### ISSUE THREAD

**Issue by Souravi785** [dazedcat19/FMD#1136]
**Prerequisite**
- [x] I'm using the latest version.
- [x] All website modules are up to date.

- [x] This is not temporary server issue.

**System information**
Windows 10-64 bit

**Describe the bug**
I've had low download speeds the whole day. Had to pause and resume constantly. At first, I thought it was a problem with asura scans website, but I noticed I am having download speeds from other sites too. 

![image](https://user-images.githubusercontent.com/24958560/201417286-b1ace102-94e7-4da1-8511-d2ef5c928f60.png)
![image](https://user-images.githubusercontent.com/24958560/201417358-7c9dde00-f280-471e-b40f-42e67458b3cd.png)

As you can see, my internet connection is working just fine.

**Comment by l-l-l-l-l-l** [comment]
The issue may be on the server side.

Whenever I download, the download speed depends on the site from which I'm downloading, and those that have anti-DDoS or anti-bot measures tend to be the slowest. If the site sets a maximum connection speed, there's nothing you can do.

Check at Options>Websites>Advanced if changing the number of connections or something works to improve the speeds (I wouldn't advise it for Flame Scans, though, since while they are not MangaDex with their bot-ban policy, given the measures they take with the URLs of their most-read series, they may temporarily block you).

**Comment by enjoe842** [comment]
Yep, I also experienced it, every download of 1 chapter must be paused and restarted so that I can download the next chapter. For a while I used cloudflare wrap and everything was running normally again even though the download speed was not as fast as without using warp.

**Comment by Stabilostick** [comment]
I know this problem too. I’ve possibly identified two reasons for it (both of which are my own doing), and they share the same underlying cause: too many simultaneous downloads.

a) My spinning hard drives, despite having a cache, have a limited I/O rate per second. During downloads, individual images are saved in parallel (number of connections x number of simultaneous chapters). Additionally, there’s the final compression process. This puts a strain on the storage system, causing downloads to write to the disk too slowly, which results in a lower download speed and timouts.

b) When too many connections are active at the same time, services like Cloudflare may throttle the speed if they detect too many requests from a single IP address. It’s also possible that, when using multiple websites as sources in parallel with muliple FMD instances, they might all be pulling data from the same underlying server, triggering rate-limiting mechanisms.

Tip: Try downloading outside the websites' peak traffic hours and limit the parallel number of connections at a time to see if that improves performance.
